### ISSUE TITLE
teams.listRepositories and teams.listRepositoriesForUser retrieve same info as repositories.list

### ISSUE THREAD

**Issue by mgs96** [MunifTanjim/node-bitbucket#14]
When I use those methods mentioned, I get the same results, I would expect those methods to retrieve the repositories of the team(s) the user is member of, not just the repositories of the user, I tried passing the role parameter with different values to repositories.list but in all cases I get the same result, am I misunderstanding something? 

Also, pagination doesn't work for listRepositories and listRepositoriesForUser.

I'm using the CDN version, but I did a quick test with the npm version and got the same results.

Thanks in advance!

**Comment by MunifTanjim** [comment]
You should look at the [doc entries](https://bitbucketjs.netlify.com/) for those methods. Also, take a look at the [official docs](https://developer.atlassian.com/bitbucket/api/2/reference/).

`teams.listRepositories` is for `/teams/{username}/repositories` and 
`teams.listRepositoriesForUser` is for `/users/{username}/repositories`

But it ultimately depends on the `username` that you pass. If you pass the same `username` you'll get the same result (if I remember it correctly). That's just how Bitbucket's APIs are implemented.

And for `repositories.list`, check [docs](https://bitbucketjs.netlify.com/#api-repositories-repositories_list). If you have the same permission for every repository on your account, you'll get the same results.

And, what do you mean by the following statement?

> pagination doesn't work for listRepositories and listRepositoriesForUser

What was you expectation and what is happening?

**Comment by MunifTanjim** [comment]
> my guess is that's how bitbucket implemented it's api.

Yes. Apparently, `https://api.bitbucket.org/2.0/teams/{username}/repositories` endpoint is not respecting the `page` and `pagelen` query parameters.

Nothing we can do about it in this library.
